vertica_python/vertica/messages/backend_messages/data_row.py

```diff
@@ -47,21 +47,17 @@ class DataRow(BackendMessage):
 
     def __init__(self, data):
         BackendMessage.__init__(self)
-        self.values = []
-        field_count = unpack('!H', data[0:2])[0]
+        field_count = unpack_from('!H', data, 0)[0]
+        self.values = [None] * field_count
         pos = 2
 
         for i in range(field_count):
             size = unpack_from('!I', data, pos)[0]
```

Comment:
One small improvement: rather than parsing size as `I`(unsigned int), parsing as `i`(int) would represent NULL as -1, which is simpler and more readable.
```suggestion
            size = unpack_from('!i', data, pos)[0]
            if size != -1:
                ...unpack_from('{0}s'.format(size), data, pos)[0]...
```

Reply:
Ah yes, I didn't even think about this, there's no reason this was parsed as unsigned when the max length of a column is <32 MB.